src/main/java/org/jboss/tm/ExtendedJBossXATerminator.java

```diff
@@ -91,8 +91,9 @@
      *                      set in the parameter. These constants are defined in javax.transaction.xa.XAResource interface
      * @return an array of Xids to recover
      * @throws XAException An error has occurred. Possible values are XAER_RMERR, XAER_RMFAIL, XAER_INVAL, and XAER_PROTO.
+     * @throws UnsupportedOperationException Operation is not supported in the context. This happens when Narayana runs on JTS.
      */
-    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException;
+    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException, UnsupportedOperationException;
```

Comment:
Currently there is the single place where the unsupported exception is thrown. But to be honest I don't understand exactly why this is called only from EJB remoting side and never for JTS.

@tomjenkinson what do you think about Mike's question, please?